src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -108,7 +104,26 @@ public function process(ContainerBuilder $container)
                 throw new InvalidArgumentException(sprintf('Service "%s" must implement interface "%s".', $id, $interface));
             }
 
-            $definition->addMethodCall('addSubscriberService', array($id, $class));
+            $r = new \ReflectionClass($class);
+            $extractingDispatcher->addSubscriber($r->newInstanceWithoutConstructor());
```

Comment:
Won't work if the `getSubscribedEvents` is somewhat configured via "static" properties or a global state. We probably don't care though.